File: app/src/main/java/me/buck/fileio/FileIoApi.java
package me.buck.fileio;

import okhttp3.MultipartBody;
import retrofit2.Call;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

/**
 * Created by gwf on 2019/5/27
 */
public interface FileIoApi {

    String BASE_URL = "https://file.io/";


    @Multipart
    @POST("/")
    Call<FileIoBean> upload(@Part MultipartBody.Part body);

}

File: app/src/main/java/me/buck/fileio/FileIoApi.kt
package me.buck.fileio

import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.*

/**
 * Created by gwf on 2019/5/30
 */
interface FileIoApi {

    companion object {
        const val BASE_URL = "https://file.io/"
    }

    @Multipart
    @POST("/")
    fun upload(@Part body: MultipartBody.Part): Call<FileIoRsp>

    @Streaming
    @GET("/{key}")
    fun download(@Path("key") key: String):Call<ResponseBody>

}